### ISSUE TITLE
One service provider

### ISSUE THREAD

**Issue by joacar** [Particular/NServiceBus#5458]
Hi,

I asked a question at [Stackoverflow](https://stackoverflow.com/questions/58391988/shared-service-provider-nservicebus-and-aspnet-core) regarding a solution how to, unobtrusively, publish integration events (using NServiceBus) from recorded domain events. 

The integration events, if any, should be published when the command executing the operation (unit of work) that affects entities that might record events. 

I published the key parts for the solution I've been working on as a [Gist](https://gist.github.com/joacar/273d84da5219bf57c8d47d1a59cff68f).

Thanks

**Comment by SzymonPobiega** [comment]
Hi @joacar 

I believe this sample should be helpful https://docs.particular.net/samples/dependency-injection/aspnetcore/

**Comment by joacar** [comment]
That does indeed look like what I want to achieve. When is 7.2 about the be released?

**Comment by SzymonPobiega** [comment]
@joacar we plan to release it early next week.

**Comment by joacar** [comment]
I implemented it the 7.2-way and it works. Not the expected result, though, since I thought that the injected class in a handler should be the same instance resolved from within the `Handle` method using the service provider created from scope.

So as I wrote this I think the reason they weren't the same is the service provider in `Handle` is scoped from a scope :) I had a behavior wrapping the pipeline with service scope

```c#
using(var scope = _serviceProviderScopeFactory.CreateScope())
{
   await next().ConfigureAwait(false);
}
```

**Comment by SzymonPobiega** [comment]
@joacar Not sure I follow. NServiceBus already crates a new scope for each message being processed. Why do you need another scope?
